### ISSUE TITLE
Last  fluid wagon on 3 car narrow stacker are not used.

### ISSUE THREAD

**Issue by toniz4** [Opinionated-Blueprints/10-Books-Full-of-Rails#136]
### Have you checked the following resources?

- [X] [Wiki](https://github.com/Opinionated-Blueprints/10-Books-Full-of-Rails/wiki)
- [X] [Discussions Forum](https://github.com/Opinionated-Blueprints/10-Books-Full-of-Rails/discussions)
- [X] [Open and Closed Issues](https://github.com/Opinionated-Blueprints/10-Books-Full-of-Rails/issues)

### Version of the Book

2.21.2

### Bug Description

The last wagon on a fluid provider are not used when the train comes from the not straight path.

### How to reproduce the problem

- create a straight narrow 3 car stacker
- make it a fluid provider

### What did you expect?

The last wagon be filled

### What happened instead?

When the train comes from the path with a turn, the last wagon will not be filled.
<img width="1274" alt="Screenshot 2024-12-28 at 15 56 02" src="https://github.com/user-attachments/assets/4ade9284-eb37-4c69-aebc-35c23252921b" />

**Comment by Lupis007** [comment]
This is not a bug with the station. This is a limitation from Factorio itself. Wagons are not able to be filled/emptied on corner tracks. The wagon in your provided screenshot is still seen as being on the non-straight track, which is causing your issue. I imagine this station does not have issues if the train come down the straight path into it.

https://wiki.factorio.com/Fluid_wagon
![image](https://github.com/user-attachments/assets/ab3af36f-edae-4a33-9d5a-5ea84dff8c31)

**Comment by toniz4** [comment]
Yup, if the train comes from the straight track it's filled as expected. Increasing the size of the station, after the stacker, by 1 track seems to fix the issue.

-------- Original Message --------

**Comment by Lupis007** [comment]
I had actually misread the issue, and yes it is with how the station is setup in the blueprint. All of the 3 car stackers are 1 rail tile too short to be used with fluid wagons. They need to be extended by 1 rail tile and the power poles moved.

Showing both train stations, with the top one being covered by the current Straight 3 car stacker, and it being a liquid station.
![image](https://github.com/user-attachments/assets/8a293660-6163-4616-8967-28a4237dd70a)

Showing both train stations again, but having the center route disconnected which, proves that it is an issue with the length of the station when it comes to liquid trains.
![image](https://github.com/user-attachments/assets/600b605f-79bf-48d6-876c-adbcdcd762df)

And this only effects fluid wagons, as solid wagons work just fine. I put 50 coal in each chest, and it grabbed it out of each just fine.
![image](https://github.com/user-attachments/assets/61cf003a-f8b1-4cc5-9f10-3b6dd2609cce)

**Comment by Lupis007** [comment]
I have updated the ones that looked like it needed it, that were not apart of Space Age since I don't currently have the expansion pack and have not felt the need to get it yet due to learning the new mechanics. 

[3CarStackerUpdated.txt](https://github.com/user-attachments/files/18271064/3CarStackerUpdated.txt)

**Comment by Lupis007** [comment]
I was doing some play testing, and noticed that the same issue crops up on the 5 car stackers. It only effects the fluid wagons, the cargo wagons are just fine.

**Comment by toniz4** [comment]
Unloading stations are still affected by this too.

<img width="531" alt="image" src="https://github.com/user-attachments/assets/5d2aaa29-f6db-4a2e-a5a8-6552255bbee9" />

**Comment by daz96050** [comment]
[2-car.txt](https://github.com/user-attachments/files/18371532/2-car.txt)
[3-car.txt](https://github.com/user-attachments/files/18371533/3-car.txt)
I fixed this issue for the 2-car and 3-car books. If you haven't done the work yet @rpdelaney
